src/components/settings-menu/NotificationsSection.tsx

```diff
@@ -212,29 +226,48 @@ const NotificationsSection = () => {
     return { ownedWallets, watchedWallets };
   }, [wallets]);
 
+  useFocusEffect(
+    useCallback(() => {
+      setOwnedState({ loading: false, status: storedOwnerEnabled });
+      setWatchedState({ loading: false, status: storedWatcherEnabled });
+    }, [storedOwnerEnabled, storedWatcherEnabled])
+  );
+
```

Comment:
The local state is always updated when we first push the screen, that means when we go to that screen from settings sheet.

Then we need to update it when we go back from single wallet settings screen, this focus effect takes care of that